experimental/packages/otlp-exporter-base/src/platform/browser/OTLPExporterBrowserBase.ts

```diff
@@ -82,7 +87,16 @@ export abstract class OTLPExporterBrowserBase<
           resolve,
           reject
         );
-      } else {
+      } else if (this.sendMethod === 'fetch') {
```

Comment:
There are lots of options for refactoring to make this smaller and to abstract this out to better support a bunch of stuff. (But that is beyond the scope you are trying to fix -- I'll probably end up addressing this in the web sandbox to prototype the change).

But a "simple" version of that would be to define a `type` which is the signature for the `send` command and just have all `implementations` (XHR, Fetch, Beacon, http) provide a matching entrypoint function matching that signature. Then during initialization (or if the `sendMethod` is changed) this `if` check would just reassign to the function and the usage would just use it's configured sending function.... (This may also be beyond the scope of this PR -- so I'm happy to delay that level of change, really just calling out (what I believe would be) a better possible design.

Reply:
I agree with you.
Such a refactor is necessary in the future.
Like sender abstraction with the retry mechanism in utils.js.